### PR TITLE
utils: new formula utils

### PR DESCRIPTION
**Pull request by kaplun**
* Introduces new LaTeX/MathML formula utilities, based on a MathJax
  server running on NodeJS.

Signed-off-by: Samuele Kaplun <samuele.kaplun@cern.ch>

### CODE REVIEW

File: inspirehep/utils/formulas.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            count -= 1
+        if count < 0:
+            return False
+    return count == 0
+
+
+def tex2mml(formula):
+    """Converts a LaTeX formula to MathML."""
+    try:
+        mathjax_node_server = current_app.config['MATHJAX_NODE_SERVER']
```
Comment (jalavik) [inline inspirehep/utils/formulas.py]: Could be simplified with
```python
current_app.config.get('MATHJAX_NODE_SERVER', MATHJAX_NODE_SERVER)
```

File: inspirehep/utils/tex2mml.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+#! /usr/bin/env node
+
+// This file is part of INSPIRE.
+// Copyright (C) 2015 CERN.
```
Comment (jalavik) [inline inspirehep/utils/tex2mml.js]: 2016

File: inspirehep/utils/tex2mml.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            format: ("inline-TeX"),
+            mml:true,
+        }, function (data) {
+            if (!data.errors) {
+                response.write(data.mml);
+            }
+        });
+    response.end();
+  });
+}).listen(48999);
```
Comment (jacquerie) [inline inspirehep/utils/tex2mml.js]: Maybe make this configurable?

File: inspirehep/utils/formulas.py (hunk 1 above)
Comment (kaplun) [inline inspirehep/utils/formulas.py]: The issue was to protect against working outside of the context.
`current_app.config.get('MATHJAX_NODE_SERVER', MATHJAX_NODE_SERVER)` would still raise `RuntimeError` outside of the context.

File: inspirehep/utils/tex2mml.js (hunk 3 above)
Comment (kaplun) [inline inspirehep/utils/tex2mml.js]: Sure. I just need to learn a bit of JS and NodeJS first :smile:

File: tests/test_formulas.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        tokens = {
+            u'Bc+',
+            u'Bc+→J/ψDs+',
+            u'ψDs∗+',
+            u'J/ψDs+',
+            u'J',
+            u'Bc+→J/ψDs∗+',
+            u'J/ψDs∗+',
+            u'ψDs+'}
+        self.assertEquals(formulas.get_all_unicode_formula_tokens_from_text(text), tokens)
```
Comment (jacquerie) [inline tests/test_formulas.py]: Deprecated, use `self.assertEqual` instead (https://docs.python.org/2/library/unittest.html#deprecated-aliases).

File: inspirehep/utils/formulas.py (hunk 1 above)
Comment (jalavik) [inline inspirehep/utils/formulas.py]: I see. Is it normal that it would run outside of the context? Do we want it to?

Comment (kaplun) [inline inspirehep/utils/formulas.py]: Probably not :-) (since it depends on an available, configurable server)

**Comment by kaplun** [comment]
Mmh. Our Docker image uses an old NodeJS. I will update it with the latest version 5.

**Pull request merged**